components-core/src/main/java/com/adyen/checkout/components/api/SuspendConnection.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2021 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by caiof on 6/1/2021.
+ */
+
+package com.adyen.checkout.components.api
+
+import com.adyen.checkout.core.api.Connection
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.supervisorScope
+import kotlinx.coroutines.withContext
+
+abstract class SuspendConnection<T>(url: String) : Connection<T>(url) {
```

Comment:
If you don't want to create a new class just for this function, you can replace it with an extension function, something like this:

```
suspend inline fun <reified T> Connection<T>.suspendedCall(): T {
    return supervisorScope {
        withContext(Dispatchers.IO) {
            call()
        }
    }
}
```

Reply:
Good idea, thanks.